tox.ini

```diff
@@ -49,10 +66,23 @@ commands =
 [testenv:build-json-schemas]
 description = build json schemas in docs/
 deps =
-    pyyaml
-    git+https://github.com/PietroPasotti/interface-tester-pytest
+    .[json_schemas]
+    # todo replace with pypi package when available
+    git+https://github.com/canonical/interface-tester-pytest
 setenv =
     PYTHONPATH={toxinidir}
 commands =
     python {[vars]docs_path}/build.py
 
+
+
+[testenv:run-interface-test-matrix]
```

Comment:
As it is, running `tox -e run-interface-test-matrix` yields the following error:

```
Traceback (most recent call last):
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/run_matrix.py", line 203, in <module>
    run_interface_tests(Path("."), include_arg)
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/run_matrix.py", line 171, in run_interface_tests
    charm_path, test_path = prepare_repo(charm_config, interface)
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/run_matrix.py", line 52, in prepare_repo
    _setup_venv(charm_path)
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/run_matrix.py", line 107, in _setup_venv
    os.chdir(charm_path)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/charm-relation-interfaces-tests/traefik-k8s'
```

Reply:
needed to remove the branch test config in ingress, now that the PR in traefik has merged.
Done.

Reply:
you should see:

![image](https://user-images.githubusercontent.com/6230162/233384284-803519bb-1514-4922-aea6-f68d5c65ad56.png)

Reply:
It's still not working for me, I'm getting this error:

```
Traceback (most recent call last):
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/run_matrix.py", line 14, in <module>
    from interface_tester.collector import collect_tests
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/.tox/run-interface-test-matrix/lib/python3.11/site-packages/interface_tester/__init__.py", line 5, in <module>
    from .interface_test import interface_test_case
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/.tox/run-interface-test-matrix/lib/python3.11/site-packages/interface_tester/interface_test.py", line 12, in <module>
    from scenario import Event, State
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/.tox/run-interface-test-matrix/lib/python3.11/site-packages/scenario/__init__.py", line 5, in <module>
    from scenario.runtime import trigger
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/.tox/run-interface-test-matrix/lib/python3.11/site-packages/scenario/runtime.py", line 28, in <module>
    from scenario.state import DeferredEvent, PeerRelation, StoredState
  File "/home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/.tox/run-interface-test-matrix/lib/python3.11/site-packages/scenario/state.py", line 689, in <module>
    @dataclasses.dataclass
     ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/dataclasses.py", line 1220, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/usr/lib/python3.11/dataclasses.py", line 1210, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/dataclasses.py", line 958, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/dataclasses.py", line 815, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'scenario.state._EntityStatus'> for field app is not allowed: use default_factory

```

Reply:
Found out it was a python 3.11 backwards-incompatible change to how dataclasses behave. Fixed in [scenario 2.2](https://github.com/canonical/ops-scenario/releases/tag/2.2)

Reply:
Thanks, good job on the fix.

I got a new problem now. While the tox command succeed, I still get a "Failed" test so:
1. Is it normal that the tests pass even if we have this issue?
2. We should probably fix the issue

Here's a log of the run.
[tox_command.txt](https://github.com/canonical/charm-relation-interfaces/files/11310609/tox_command.txt)